test/helpers/assets.ts

```diff
@@ -0,0 +1,172 @@
+import "@moonbeam-network/api-augment/moonbase";
+import { u128 } from "@polkadot/types";
+import { BN, hexToU8a, u8aToHex } from "@polkadot/util";
+import { expect, DevModeContext } from "@moonwall/cli";
+import { ApiPromise } from "@polkadot/api";
+import { blake2AsU8a, xxhashAsU8a } from "@polkadot/util-crypto";
+import { KeyringPair } from "@polkadot/keyring/types";
+import type { PalletAssetsAssetAccount, PalletAssetsAssetDetails } from "@polkadot/types/lookup";
+import type { AccountId20 } from "@polkadot/types/interfaces/runtime";
+
+export const DUMMY_REVERT_BYTECODE = "0x60006000fd";
+export const RELAY_SOURCE_LOCATION = { Xcm: { parents: 1, interior: "Here" } };
+export const RELAY_SOURCE_LOCATION2 = { Xcm: { parents: 2, interior: "Here" } };
+export const RELAY_V3_SOURCE_LOCATION = { V3: { parents: 1, interior: "Here" } } as any;
+export const PARA_1000_SOURCE_LOCATION = {
+  Xcm: { parents: 1, interior: { X1: { Parachain: 1000 } } },
+};
+export const PARA_2000_SOURCE_LOCATION = {
+  Xcm: { parents: 1, interior: { X1: { Parachain: 2000 } } },
+};
+export const PARA_1001_SOURCE_LOCATION = {
+  Xcm: { parents: 1, interior: { X1: { Parachain: 1001 } } },
+};
+
+export interface AssetMetadata {
+  name: string;
+  symbol: string;
+  decimals: BN;
+  isFrozen: boolean;
+}
+
+export const relayAssetMetadata: AssetMetadata = {
+  name: "DOT",
+  symbol: "DOT",
+  decimals: new BN(12),
+  isFrozen: false,
+};
+
+export async function mockAssetBalance(
+  context: DevModeContext,
+  assetBalance: PalletAssetsAssetAccount,
+  assetDetails: PalletAssetsAssetDetails,
+  sudoAccount: KeyringPair,
+  assetId: u128,
+  account: string | AccountId20,
+  is_sufficient = false
+) {
+  const api = context.polkadotJs({ type: "moon" });
+  // Register the asset
+  await context.createBlock(
+    api.tx.sudo
+      .sudo(
+        api.tx.assetManager.registerForeignAsset(
+          RELAY_SOURCE_LOCATION,
+          relayAssetMetadata,
+          new BN(1),
+          is_sufficient
+        )
+      )
+      .signAsync(sudoAccount)
+  );
+
+  const assets = await api.query.assetManager.assetIdType(assetId);
+  // make sure we created it
+  expect(assets.unwrap().asXcm.parents.toNumber()).to.equal(1);
+
+  // Get keys to modify balance
+  let module = xxhashAsU8a(new TextEncoder().encode("Assets"), 128);
```

Comment:
I'd suggest to use "const" keyword for immutable variables. This makes it easier to read